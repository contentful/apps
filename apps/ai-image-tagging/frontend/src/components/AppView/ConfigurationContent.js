import React from 'react';
import { List, ListItem, Heading, Paragraph } from '@contentful/forma-36-react-components';
import { Divider } from '../Divider';
import { styles } from './styles';

export const ConfigurationContent = () => (
  <>
    <Heading className={styles.heading}>Getting started</Heading>
    <Paragraph>Follow the next steps to create a new image entry with AI generated tags:</Paragraph>
    <List className={styles.list}>
      <ListItem>Go to the &quot;Content&quot; page</ListItem>
      <ListItem>
        Create a new entry of type &quot;AI Image Tagging&quot; (or the name you chose during
        the installation)
      </ListItem>
      <ListItem>Fill in the required fields and publish</ListItem>
    </List>
    <Paragraph className={styles.paragraph}>
      Follow the next steps to reference the newly created image with AI Image Tagging in another
      content type:
    </Paragraph>
    <List className={styles.list}>
      <ListItem>Go to the &quot;content model&quot; page</ListItem>
      <ListItem>
        Edit the content type that needs to reference the image with the AI tags
      </ListItem>
      <ListItem>Create a new field of type &quot;reference&quot;</ListItem>
      <ListItem>
        Set a validation rule requiring that the content type the reference points to is of type
        &quot;AI Image Tagging&quot;
      </ListItem>
      <ListItem>
        Optionally set the appearance of the new reference field to &quot;entry card&quot; in its
        settings
      </ListItem>
      <ListItem>Go back to the &quot;content&quot; page</ListItem>
      <ListItem>
        Create/edit an entry of the content type referencing the image with the AI tags
      </ListItem>
      <ListItem>
        Set the reference field you added to point to the image with AI tags entry
      </ListItem>
    </List>
    <Divider />
    <Heading className={styles.heading}>Uninstallation and cleanup</Heading>
    <Paragraph>
      If you uninstall the AI Image Tagging app you will have to manually clean up the content type
      we created for you during the installation. To do that follow the next steps:
    </Paragraph>
    <List className={styles.list}>
      <ListItem>Go to the &quot;content&quot; page</ListItem>
      <ListItem>
        Delete any entries of type &quot;AI Image Tagging&quot; (or the name you chose during
        installation)
      </ListItem>
      <ListItem>Go to the &quot;content model&quot; page</ListItem>
      <ListItem>
        Edit the content type the app created, select &quot;actions&quot; from the top menu and
        &quot;delete&quot;
      </ListItem>
    </List>
  </>
);
