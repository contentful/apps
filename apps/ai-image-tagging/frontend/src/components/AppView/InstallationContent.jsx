import React from 'react';
import PropTypes from 'prop-types';
import { Heading, Paragraph, FormControl, Flex, TextInput } from '@contentful/f36-components';
import { styles } from './styles';

export function InstallationContent({
  allContentTypesIds,
  contentTypeId,
  contentTypeName,
  onContentTypeNameChange,
  onContentTypeIdChange,
}) {
  const validationMessageId = allContentTypesIds.includes(contentTypeId)
    ? `A content type with ID "${contentTypeId}" already exists. Try a different ID.`
    : null;

  return (
    <>
      <Heading className={styles.heading}>Configuration</Heading>
      <Paragraph>
        To help you get started, we are going to create a content type for you with a title field,
        an image field and a image tags field.
      </Paragraph>
      <FormControl id="content-type-name" testId="content-type-name" className={styles.input}>
        <FormControl.Label>Content type name</FormControl.Label>
        <TextInput
          name="contentTypeName"
          placeholder="e.g. AI Tagged Image"
          testId="content-type-name-input"
          value={contentTypeName}
          onChange={onContentTypeNameChange}
          required
        />
        <Flex justifyContent="space-between">
          <FormControl.HelpText>
            You can use this content type to add tags to images
          </FormControl.HelpText>
        </Flex>
      </FormControl>
      <FormControl id="content-type-id" testId="content-type-id" className={styles.input}>
        <FormControl.Label>Content type ID</FormControl.Label>
        <TextInput
          name="contentTypeId"
          value={contentTypeId}
          onChange={onContentTypeIdChange}
          testId="content-type-id-input"
          required
        />
        <Flex justifyContent="space-between">
          <FormControl.HelpText>
            The ID is generated from the name, you can also set it manually
          </FormControl.HelpText>
        </Flex>
        <FormControl.ValidationMessage>{validationMessageId}</FormControl.ValidationMessage>
      </FormControl>
    </>
  );
}

InstallationContent.propTypes = {
  allContentTypesIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  contentTypeId: PropTypes.string.isRequired,
  contentTypeName: PropTypes.string.isRequired,
  onContentTypeNameChange: PropTypes.func.isRequired,
  onContentTypeIdChange: PropTypes.func.isRequired,
};
