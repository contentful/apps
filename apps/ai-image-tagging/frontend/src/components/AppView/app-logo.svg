<svg height="30" viewBox="0 0 30 30" width="30" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><linearGradient id="a" x1="50%" x2="50%" y1="0%" y2="100%"><stop offset="0" stop-color="#fcaa11"/><stop offset="1" stop-color="#ea9005"/></linearGradient><g fill="none" fill-rule="evenodd"><rect fill="url(#a)" fill-rule="nonzero" height="30" rx="3" width="30"/><g stroke="#fff" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"><g transform="translate(3 5)"><path d="m23.5 3.5v-3h-3"/><path d="m23.5 16.5v3h-3"/><path d="m.5 3.5v-3h3"/><path d="m.5 16.5v3h3"/></g><g transform="translate(10.5 10.5)"><path d="m0 4.5h9"/><path d="m0 4.5h9" transform="matrix(0 1 -1 0 9 0)"/></g></g></g></svg>
