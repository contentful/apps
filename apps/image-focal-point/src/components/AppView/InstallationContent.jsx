import React from 'react';
import PropTypes from 'prop-types';
import { Heading, Paragraph, FormControl, TextInput, Flex } from '@contentful/f36-components';
import { styles } from './styles';

export function InstallationContent({
  allContentTypesIds,
  contentTypeId,
  contentTypeName,
  onContentTypeNameChange,
  onContentTypeIdChange,
}) {
  const validationMessageName = allContentTypesIds.includes(contentTypeId)
    ? `A content type with ID "${contentTypeId}" already exists. Try a different name.`
    : null;

  const validationMessageId = allContentTypesIds.includes(contentTypeId)
    ? `A content type with ID "${contentTypeId}" already exists. Try a different ID.`
    : null;

  return (
    <>
      <Heading className={styles.heading}>Configuration</Heading>
      <Paragraph>
        To help you get started, we are going to create a content type for you with a title field,
        an image field and a focal point field.
      </Paragraph>
      <FormControl id="content-type-name" testId="content-type-name">
        <FormControl.Label>Content type name</FormControl.Label>
        <TextInput
          placeholder="e.g. Image with Focal Point"
          name="contentTypeName"
          testId="content-type-name-input"
          value={contentTypeName}
          onChange={onContentTypeNameChange}
          className={styles.input}
          required
        />
        <Flex justifyContent="space-between">
          <FormControl.HelpText>
            You can use this content type to wrap images with focal point data
          </FormControl.HelpText>
          <FormControl.Counter />
        </Flex>
        <FormControl.ValidationMessage>{validationMessageName}</FormControl.ValidationMessage>
      </FormControl>
      <FormControl id="content-type-id" testId="content-type-id">
        <FormControl.Label>Content type ID</FormControl.Label>
        <TextInput
          name="contentTypeId"
          testId="content-type-id-input"
          value={contentTypeId}
          onChange={onContentTypeIdChange}
          className={styles.input}
          required
        />
        <Flex justifyContent="space-between">
          <FormControl.HelpText>
            The ID is generated from the name, you can also set it manually
          </FormControl.HelpText>
          <FormControl.Counter />
        </Flex>
        <FormControl.ValidationMessage>{validationMessageId}</FormControl.ValidationMessage>
      </FormControl>
    </>
  );
}

InstallationContent.propTypes = {
  allContentTypesIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  contentTypeId: PropTypes.string.isRequired,
  contentTypeName: PropTypes.string.isRequired,
  onContentTypeNameChange: PropTypes.func.isRequired,
  onContentTypeIdChange: PropTypes.func.isRequired,
};
