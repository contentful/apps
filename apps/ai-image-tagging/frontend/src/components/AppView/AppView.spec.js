import React from 'react';
import { render, waitFor, configure, fireEvent } from '@testing-library/react';

import mockProps from '../../test/mockProps';
import { AppView } from './AppView';

configure({ testIdAttribute: 'data-test-id' });

describe('AppView', () => {
  let props;
  beforeEach(() => {
    props = {
      sdk: {
        ...mockProps.sdk,
        space: {
          getContentTypes: jest.fn(() => Promise.resolve({ items: [] })),
        },
        app: {
          setReady: jest.fn(),
          isInstalled: jest.fn(),
          onConfigure: jest.fn(),
          onConfigurationCompleted: jest.fn(),
        },
      },
    };
  });

  describe('when the app is not installed', () => {
    beforeEach(() => props.sdk.app.isInstalled.mockImplementation(() => Promise.resolve(false)));

    it('should render the app view with installation screen', async () => {
      const appView = render(<AppView {...props} />);
      await waitFor(() => appView.getByText('Configuration'));
      expect(appView.container).toMatchSnapshot();
    });

    it('should render inline validation if the content type id is taken', async () => {
      props.sdk.space.getContentTypes.mockImplementation(() =>
        Promise.resolve({
          items: [{ sys: { id: 'imageWithFocalPoint' } }],
        })
      );
      const { getByTestId, getByText } = render(<AppView {...props} />);
      await waitFor(() => getByText('Configuration'));
      expect(getByTestId('content-type-name')).toMatchSnapshot();
      expect(getByTestId('content-type-id')).toMatchSnapshot();
    });

    it('should update the content type id field if it is pristine and the content type name is changed', async () => {
      const { getByTestId, getByText } = render(<AppView {...props} />);
      await waitFor(() => getByText('Configuration'));
      fireEvent.change(getByTestId('content-type-name-input'), { target: { value: 'Test Name' } });
      expect(getByTestId('content-type-id-input').value).toEqual('testName');
    });

    it('should not update the content type id field if it is dirty and the content type name is changed', async () => {
      const { getByTestId, getByText } = render(<AppView {...props} />);
      await waitFor(() => getByText('Configuration'));
      fireEvent.change(getByTestId('content-type-id-input'), { target: { value: 'someTestId' } });
      fireEvent.change(getByTestId('content-type-name-input'), { target: { value: 'Test Name' } });
      expect(getByTestId('content-type-id-input').value).toEqual('someTestId');
    });
  });

  describe('when the app is installed', () => {
    it('should render the app view with configuration screen', async () => {
      props.sdk.app.isInstalled.mockImplementation(() => true);
      const appView = render(<AppView {...props} />);
      await waitFor(() => appView.getByText('Configuration'));
      expect(appView.container).toMatchSnapshot();
    });
  });
});
