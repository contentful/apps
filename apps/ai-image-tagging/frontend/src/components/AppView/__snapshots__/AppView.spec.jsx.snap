// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppView > when the app is installed > should render the app view with configuration screen 1`] = `
<div>
  <div
    class="css-7bfeg1"
  />
  <div
    class="css-jwwzlh"
  >
    <h1
      class="css-acwcvw css-ow69k3"
      data-test-id="cf-ui-heading"
    >
      About AI Image Tagging
    </h1>
    <p
      class="css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      The AI Image Tagging app uses AI and deep learning capabilities to automatically assign tags to images. The tags are visible and accessible via the Contentful web app entry editor and tags are searchable in the search bar.
    </p>
    <hr
      class="css-10hsrou"
    />
    <h1
      class="css-acwcvw css-ow69k3"
      data-test-id="cf-ui-heading"
    >
      Getting started
    </h1>
    <p
      class="css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      Follow the next steps to create a new image entry with AI generated tags:
    </p>
    <ul
      class="css-1uk1gs8 css-qslnu8"
      data-test-id="cf-ui-list"
    >
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Go to the "Content" page
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Create a new entry of type "AI Image Tagging" (or the name you chose during the installation)
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Fill in the required fields and publish
      </li>
    </ul>
    <p
      class="css-15g2oxy css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      Follow the next steps to reference the newly created image with AI Image Tagging in another content type:
    </p>
    <ul
      class="css-1uk1gs8 css-qslnu8"
      data-test-id="cf-ui-list"
    >
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Go to the "content model" page
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Edit the content type that needs to reference the image with the AI tags
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Create a new field of type "reference"
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Set a validation rule requiring that the content type the reference points to is of type "AI Image Tagging"
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Optionally set the appearance of the new reference field to "entry card" in its settings
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Go back to the "content" page
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Create/edit an entry of the content type referencing the image with the AI tags
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Set the reference field you added to point to the image with AI tags entry
      </li>
    </ul>
    <hr
      class="css-10hsrou"
    />
    <h1
      class="css-acwcvw css-ow69k3"
      data-test-id="cf-ui-heading"
    >
      Uninstallation and cleanup
    </h1>
    <p
      class="css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      If you uninstall the AI Image Tagging app you will have to manually clean up the content type we created for you during the installation. To do that follow the next steps:
    </p>
    <ul
      class="css-1uk1gs8 css-qslnu8"
      data-test-id="cf-ui-list"
    >
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Go to the "content" page
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Delete any entries of type "AI Image Tagging" (or the name you chose during installation)
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Go to the "content model" page
      </li>
      <li
        class="css-gic5lh"
        data-test-id="cf-ui-list-item"
      >
        Edit the content type the app created, select "actions" from the top menu and "delete"
      </li>
    </ul>
  </div>
  <div
    class="css-13jvbbd"
  >
    <img
      alt="logo"
      src="/src/components/AppView/app-logo.svg"
    />
  </div>
</div>
`;

exports[`AppView > when the app is not installed > should render inline validation if the content type id is taken 1`] = `
<div
  class="css-15g2oxy css-1fobf8d"
  data-test-id="content-type-name"
  role="group"
>
  <label
    class="css-1ry5edl"
    data-test-id="cf-ui-form-label"
    for="content-type-name"
    id="content-type-name-label"
  >
    Content type name
  </label>
  <input
    aria-describedby="content-type-name-helptext"
    class="css-82ji86"
    data-test-id="content-type-name-input"
    id="content-type-name"
    name="contentTypeName"
    placeholder="e.g. AI Tagged Image"
    type="text"
    value="Image with AI tags"
  />
  <div
    class="css-gg4vpm"
  >
    <p
      class="css-1l2c7z"
      data-test-id="cf-ui-help-text"
      id="content-type-name-helptext"
    >
      You can use this content type to add tags to images
    </p>
  </div>
</div>
`;

exports[`AppView > when the app is not installed > should render inline validation if the content type id is taken 2`] = `
<div
  class="css-15g2oxy css-1fobf8d"
  data-test-id="content-type-id"
  role="group"
>
  <label
    class="css-1ry5edl"
    data-test-id="cf-ui-form-label"
    for="content-type-id"
    id="content-type-id-label"
  >
    Content type ID
  </label>
  <input
    aria-describedby="content-type-id-helptext"
    class="css-82ji86"
    data-test-id="content-type-id-input"
    id="content-type-id"
    name="contentTypeId"
    type="text"
    value="imageWithAiTags"
  />
  <div
    class="css-gg4vpm"
  >
    <p
      class="css-1l2c7z"
      data-test-id="cf-ui-help-text"
      id="content-type-id-helptext"
    >
      The ID is generated from the name, you can also set it manually
    </p>
  </div>
  <div
    aria-live="assertive"
    class="css-dcyvog"
    data-test-id="cf-ui-validation-message"
    id="content-type-id-validation"
  >
    <div
      class="css-ql15kp"
    >
      <svg
        aria-hidden="true"
        class="css-cvj358"
        data-test-id="cf-ui-icon"
        role="img"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <p
      class="css-1uhwrh7"
    />
  </div>
</div>
`;

exports[`AppView > when the app is not installed > should render the app view with installation screen 1`] = `
<div>
  <div
    class="css-7bfeg1"
  />
  <div
    class="css-jwwzlh"
  >
    <h1
      class="css-acwcvw css-ow69k3"
      data-test-id="cf-ui-heading"
    >
      About AI Image Tagging
    </h1>
    <p
      class="css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      The AI Image Tagging app uses AI and deep learning capabilities to automatically assign tags to images. The tags are visible and accessible via the Contentful web app entry editor and tags are searchable in the search bar.
    </p>
    <hr
      class="css-10hsrou"
    />
    <h1
      class="css-acwcvw css-ow69k3"
      data-test-id="cf-ui-heading"
    >
      Configuration
    </h1>
    <p
      class="css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      To help you get started, we are going to create a content type for you with a title field, an image field and a image tags field.
    </p>
    <div
      class="css-15g2oxy css-1fobf8d"
      data-test-id="content-type-name"
      role="group"
    >
      <label
        class="css-1ry5edl"
        data-test-id="cf-ui-form-label"
        for="content-type-name"
        id="content-type-name-label"
      >
        Content type name
      </label>
      <input
        aria-describedby="content-type-name-helptext"
        class="css-82ji86"
        data-test-id="content-type-name-input"
        id="content-type-name"
        name="contentTypeName"
        placeholder="e.g. AI Tagged Image"
        type="text"
        value="Image with AI tags"
      />
      <div
        class="css-gg4vpm"
      >
        <p
          class="css-1l2c7z"
          data-test-id="cf-ui-help-text"
          id="content-type-name-helptext"
        >
          You can use this content type to add tags to images
        </p>
      </div>
    </div>
    <div
      class="css-15g2oxy css-1fobf8d"
      data-test-id="content-type-id"
      role="group"
    >
      <label
        class="css-1ry5edl"
        data-test-id="cf-ui-form-label"
        for="content-type-id"
        id="content-type-id-label"
      >
        Content type ID
      </label>
      <input
        aria-describedby="content-type-id-helptext"
        class="css-82ji86"
        data-test-id="content-type-id-input"
        id="content-type-id"
        name="contentTypeId"
        type="text"
        value="imageWithAiTags"
      />
      <div
        class="css-gg4vpm"
      >
        <p
          class="css-1l2c7z"
          data-test-id="cf-ui-help-text"
          id="content-type-id-helptext"
        >
          The ID is generated from the name, you can also set it manually
        </p>
      </div>
      <div
        aria-live="assertive"
        class="css-dcyvog"
        data-test-id="cf-ui-validation-message"
        id="content-type-id-validation"
      >
        <div
          class="css-ql15kp"
        >
          <svg
            aria-hidden="true"
            class="css-cvj358"
            data-test-id="cf-ui-icon"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <p
          class="css-1uhwrh7"
        />
      </div>
    </div>
  </div>
  <div
    class="css-13jvbbd"
  >
    <img
      alt="logo"
      src="/src/components/AppView/app-logo.svg"
    />
  </div>
</div>
`;
